modules/thermo-cycler/thermo-cycler-arduino/lid.cpp

```diff
@@ -46,28 +46,76 @@ void Lid::_setup_digipot()
 
 Lid_status Lid::status()
 {
-  if(_is_open_switch_pressed() && !_is_closed_switch_pressed())
+  return _status;
+}
+
+void Lid::_update_status()
+{
+  uint8_t status_bits = (_is_cover_switch_pressed << 1) | (_is_bottom_switch_pressed << 0);
+  /* Truth Table */
+  /* Bit 1 | Bit 0
+  /*    0     0   -- Both switches OPEN
+   *    0     1   -- Bottom switch CLOSED, cover switch OPEN
+   *    1     0   -- Bottom switch OPEN, cover switch CLOSED
+   *    1     1   -- Both switches CLOSED
+   */
+  switch(status_bits)
   {
-    return Lid_status::open;
+    case 0x00:
+      _status = Lid_status::in_between;
+      break;
+    case 0x01:
+      _status = Lid_status::closed;
+      break;
+    case 0x02:
+      _status = Lid_status::open;
+      break;
+    case 0x03:
+      _status = Lid_status::error;
+      break;
   }
-  else if(_is_closed_switch_pressed() && !_is_open_switch_pressed())
+}
+
+/* This poller keeps track of changes on the lid switches and updating the status.
+ * Uses a bit shifting technique described here:
+ * https://forum.arduino.cc/index.php?topic=451164.0
+ */
+void Lid::switch_poller()
```

Comment:
There's a lot of logic here and I wonder if it might be simpler to set up these pins as edge-triggered interrupts and do the debouncing part in the main loop rather than a timer interrupt.